bin/check_h5_plugin.cpp

```diff
@@ -1,29 +1,88 @@
 // SPDX-License-Identifier: MIT
 
 #include <dectris/neggia/plugin/H5ToXds.h>
+#include <chrono>
+#include <ctime>
+#include <iomanip>
+#include <ios>
 #include <iostream>
 #include <memory>
 
+#include <dlfcn.h>
+
+typedef void (*dl_plugin_open)(const char*,
+                               int info_array[1024],
+                               int* error_flag);
+
+typedef void (*dl_plugin_get_header)(int* nx,
+                                     int* ny,
+                                     int* nbytes,
+                                     float* qx,
+                                     float* qy,
+                                     int* number_of_frames,
+                                     int info_array[1024],
+                                     int* error_flag);
+
+typedef void (*dl_plugin_get_data)(int* frame_number,
+                                   int* nx,
+                                   int* ny,
+                                   int data_array[],
+                                   int info_array[1024],
+                                   int* error_flag);
+
+typedef void (*dl_plugin_close)(int* error_flag);
```

Comment:
Instead of producing so much duplicated code. You could make this more maintainable with:
```c++
using dl_plugin_open = decltype(&plugin_open);
using dl_plugin_get_header = decltype(&plugin_get_header);
using dl_plugin_get_data = decltype(&plugin_get_data);
using dl_plugin_close  = decltype(&plugin_close);
```